### PR TITLE
iperf3d: init at 1.0.0

### PR DESCRIPTION
**Pull request by NetaliDev**
###### Description of changes

Adds [iperf3d](https://github.com/wobcom/iperf3d) at 1.0.0.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by clerie** [comment]
Result of `nixpkgs-review pr 241947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>iperf3d</li>
  </ul>
</details>

File: pkgs/tools/networking/iperf3d/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "iperf3d";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "wobcom";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-pMwGoBgFRVY+H51k+YCamzHgBoaJVwEVqY0CvMPvE0w=";
+  };
+
```
Comment (clerie) [inline pkgs/tools/networking/iperf3d/default.nix:13]: Shouldn't `iperf3` be a dependency of `iperf3d`. I't does not work otherwise anyway.

**Review by clerie — CHANGES_REQUESTED** [review]
Looks good to me overall, is building and usable when adding `iperf3` to path:

```
[clerie@krypton pr-241947]$ iperf3d --server
Listening on: [::]:6201
Dynamic port range: 7000-7999
thread 'tokio-runtime-worker' panicked at 'Failed to spawn iperf3 server: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/server.rs:287:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
^C
[clerie@krypton pr-241947]$ nix shell nixpkgs#iperf3
[clerie@krypton pr-241947]$ iperf3d --server
Listening on: [::]:6201
Dynamic port range: 7000-7999
-----------------------------------------------------------
Server listening on 7000 (test #1)
-----------------------------------------------------------
Accepted connection from ::1, port 50494
[  5] local ::1 port 7000 connected to ::1 port 50504
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-1.00   sec  4.70 GBytes  40.4 Gbits/sec                  
[  5]   1.00-2.00   sec  4.76 GBytes  40.9 Gbits/sec                  
[  5]   2.00-3.00   sec  4.91 GBytes  42.1 Gbits/sec                  
[  5]   3.00-4.00   sec  4.60 GBytes  39.5 Gbits/sec                  
[  5]   4.00-5.00   sec  4.73 GBytes  40.7 Gbits/sec                  
[  5]   5.00-6.00   sec  4.65 GBytes  39.9 Gbits/sec                  
[  5]   6.00-7.00   sec  4.59 GBytes  39.4 Gbits/sec                  
[  5]   7.00-8.00   sec  4.33 GBytes  37.2 Gbits/sec                  
[  5]   8.00-9.00   sec  4.35 GBytes  37.4 Gbits/sec                  
[  5]   9.00-10.00  sec  4.70 GBytes  40.4 Gbits/sec                  
[  5]  10.00-10.00  sec  4.19 MBytes  42.2 Gbits/sec                  
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-10.00  sec  46.3 GBytes  39.8 Gbits/sec                  receiver
^C
[clerie@krypton pr-241947]$ 
```

File: pkgs/tools/networking/iperf3d/default.nix (hunk 1 above)
Comment (NetaliDev) [reply pkgs/tools/networking/iperf3d/default.nix:13]: I have now added a wrapper which ensures that `iperf3` is in the `PATH` of `iperf3d`. Can you please try it again and confirm that this fix works for you?

**Comment by clerie** [comment]
Result of `nixpkgs-review pr 241947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>iperf3d</li>
  </ul>
</details>

**Review by clerie — APPROVED** [review]
Looks fine now. Thank you for your submission!

**Pull request merged**